types/overloads.d.ts

```diff
@@ -0,0 +1,661 @@
+// This file is generated, do not edit.
+import { EventEmitter } from "events";
+
+declare class ObsWebSocket extends EventEmitter {
```

Comment:
This looks solid.
Looks like events would be the next big task.

Reply:
Events _should_ already be handled I think? They're at the bottom of the file, after the requests.

Reply:
Whoops, missed those entirely.

I wonder if there's a way to warn when someone uses a non-recognized event or request. Technically they can be running a custom plugin, so they could be using their own names.

Reply:
Yeah I'm not sure what the best way to handle this is. Maybe @hoishin knows?

Reply:
FWIW, I absolutely want warnings when using a non-recognized event, but I agree that we need a way to allow users to use custom events.

Reply:
Does this type definition not output errors when using wrong event/requestType?

Also users should be able to extend this type definition with declaration merging.

Reply:
@Hoishin It does not currently output errors when using wrong event/requestType. I have added 2 failing type tests for this case:

```ts
// $ExpectError
obs.send('MadeUpRequestName');

// $ExpectError
obs.on('MadeUpEventName', () => {});
```

Reply:
Nevermind, I was testing the wrong file. It does seem to work after my latest commit!

Reply:
Part of that is because it extends EventEmitter, which defines `.on`